packages/components/src/Popover/Layout/PopoverHeader/PopoverHeader.tsx

```diff
@@ -50,19 +52,27 @@ type DetailOptions = WithDetail | WithHideClose
 
 export type PopoverHeaderProps = DetailOptions & {
   children: ReactNode
+  headerVisuallyHidden?: boolean
 }
 
 const PopoverHeaderLayout: FC<PopoverHeaderProps> = ({
   children,
   hideClose = false,
   detail,
+  headerVisuallyHidden = false,
   ...props
 }) => {
-  return (
+  const { id: dialogId } = useContext(DialogContext)
+  const headingId = dialogId || undefined
```

Comment:
The `heading` needs to have `-heading` prepended to it so that the `labelledby` on the "surface" can target it.

```suggestion
    const { id } = useContext(DialogContext)
    const headingId = id ? `${id}-heading` : undefined

```